task07.rb

```diff
@@ -0,0 +1,24 @@
+# Output how many square roots are between numbers x and y, e.g. if x=8 and y=37 it should output "4" for 9, 16, 25 and 36
+
+def task07(arr)
+  raise ArgumentError, "Not a valid format. Need an array" if arr.class != Array
+  raise ArgumentError, "Array must contain 2 numbers" if arr.length != 2
+  arr.each do |i|
+    raise ArgumentError, "Array must contain 2 numbers" if !(i.class == Integer || i.class == Float)
+  end
+  raise ArgumentError, "Second number must be positive or zero" if arr[1] < 0
+  raise ArgumentError, "First number can not be bigger than second." if arr[0] > arr[1]
+
+  def is_square?(x)
```

Comment:
@MaraEfimova Please do not define methods into another methods, especially when it is not needed to be inside the method.